lib/internal/url.js

```diff
@@ -840,138 +707,72 @@ class URL {
     return this[context].password;
   }
 
-  set password(password) {
+  set password(value) {
     if (!isURLThis(this))
       throw new ERR_INVALID_THIS('URL');
-    // toUSVString is not needed.
-    password = `${password}`;
-    if (this[cannotHaveUsernamePasswordPort])
-      return;
-    const ctx = this[context];
-    if (password === '') {
-      ctx.password = '';
-      ctx.flags &= ~URL_FLAGS_HAS_PASSWORD;
-      return;
-    }
-    ctx.password = encodeAuth(password);
-    ctx.flags |= URL_FLAGS_HAS_PASSWORD;
+    updateUrl(this[context].href, updateActions.kPassword, `${value}`, FunctionPrototypeBind(onParseComplete, this));
```

Comment:
Rather than using `FunctionPrototypeBind` repeatedly could these just use something like

```suggestion
   updateUrl(this[context].href, updateActions.kPassword, `${value}`, () => this.#onParseComplete);
```

And move the `onParseComplete` into a private function on `this`?

Reply:
(non-blocking, btw)

Reply:
Private methods are not writable as it seems. Resolving this issue after talking with @jasnell. For future reference: https://github.com/tc39/proposal-private-methods/issues/11

Reply:
Can use private fields still